olcUTIL_Geometry2D.h

```diff
@@ -1231,26 +1231,42 @@ namespace olc::utils::geom2d
 	template<typename T1, typename T2>
 	inline std::vector<olc::v_2d<T2>> intersects(const rect<T1>& r1, const rect<T2>& r2)
 	{
-		// TODO:
-		return {};
+		std::vector<olc::v_2d<T2>> ret;
+		for (size_t i = 0; i < 4; i++) {
+			for (size_t j = 0; j < 4; j++) {
+				auto v = intersects(r1.side(i), r2.side(j));
+				ret.insert(ret.end(), v.begin(), v.end());
+			}
+		}
+		return ret;
 	}
 
 	// intersects(c,r)
 	// Get intersection points where circle intersects with rectangle
 	template<typename T1, typename T2>
 	inline std::vector<olc::v_2d<T2>> intersects(const circle<T1>& c, const rect<T2>& r)
 	{
-		// TODO:
-		return {};
+		std::vector<olc::v_2d<T2>> ret;
+		for (size_t i = 0; i < 4; i++) {
+			auto v = intersects(c, r.side(i));
+			ret.insert(ret.end(), v.begin(), v.end());
+		}
+		return ret;
 	}
 
 	// intersects(t,r)
 	// Get intersection points where triangle intersects with rectangle
 	template<typename T1, typename T2>
 	inline std::vector<olc::v_2d<T2>> intersects(const triangle<T1>& t, const rect<T2>& r)
 	{
-		// TODO:
-		return {};
+		std::vector<olc::v_2d<T2>> ret;
+		for (size_t i = 0; i < 3; i++) {
+			for (size_t j = 0; j < 4; j++) {
+				auto v = intersects(t.side(i), r.side(j));
+				ret.insert(ret.end(), v.begin(), v.end());
+			}
+		}
+		return ret;
 	}

```

Comment:
This returns 2 intersections on corners.
Probably not desirable.
https://imgur.com/a/qKcsRWI